addons/account/models/account_bank_statement.py

```diff
@@ -915,11 +925,29 @@ def process_reconciliation(self, counterpart_aml_dicts=None, payment_aml_rec=Non
             if abs(total)>0.00001:
                 partner_id = self.partner_id and self.partner_id.id or False
                 partner_type = False
-                if partner_id:
-                    if total < 0:
-                        partner_type = 'supplier'
-                    else:
-                        partner_type = 'customer'
+                # Check based on account type whether we have a payment for receivable (customer) or payable (supplier) accounts and fallback in case none of them are used
+                if partner_id and partner_id != self.env.user.company_id.partner_id:
+                    for account in partner_accounts:
+                        block = False
+                        if account.user_type_id.id == self.env.ref('account.data_account_type_payable').id:
+                            if partner_type == 'customer':
+                                block = True
+                            partner_type = 'supplier'
+                        elif account.user_type_id.id == self.env.ref('account.data_account_type_receivable').id:
```

Comment:
you don't need to add .id, `account.user_type_id == self.env.ref('account.data_account_type_receivable')` is enought